### ISSUE TITLE
MDC Select: Inconsistent styles across browsers for CSS-only multi-select

### ISSUE THREAD

**Issue by acdvorak** [material-components/material-components-web#1751]
### What MDC-Web Version are you using?

0.27.0

### What browser(s) is this bug affecting?

Safari 11.0.1 (desktop)

### What OS are you using?

macOS Sierra 10.12.6

### What are the steps to reproduce the bug?

1. View the [`mdc-select` demo page](https://material-components-web.appspot.com/select.html) in both Safari and Chrome
2. Scroll down to "Select Multiple - CSS Only" section

### What is the expected behavior?

The `height` of each `<option class="mdc-list-item">` element should be `48px` in all browsers.

### What is the actual behavior?

Safari appears to ignore our `height` and `padding` values on `<option class="mdc-list-item">` elements. As a result, they are rendered at a much shorter `height` than in other browsers, which deviates from spec.

### Any other information you believe would be useful?

It looks like Safari isn't respecting `-moz-appearance: none` on `.mdc-multi-select`.

Screenshot of Chrome vs. Safari:
![screenshot image](https://user-images.githubusercontent.com/409245/33914730-6ccc5cac-df54-11e7-8ecf-fb71d55fb7a2.png)

**Comment by acdvorak** [comment]
Firefox also appears to style the `label` on an `<optgroup>` with italics by default:

![Screenshot of Chrome vs. Firefox](https://user-images.githubusercontent.com/409245/33966873-02550c3a-e016-11e7-8ff3-de4a1cfaffe5.png)

**Comment by kfranqueiro** [comment]
While we may have control over the italics, we don't have control over padding/margin in Safari at all. Some of this is a result of inconsistent cross-browser behavior in what browsers do/don't allow us to customize over native selects.

Another example is how Chrome overrides the selection color when the multiselect is focused, using the system selection color instead (at least on OS X).

We discussed this with design yesterday, and we ultimately want to aim to remove the native multiselect example, given the following factors:

* The aforementioned cross-browser inconsistencies beyond our control
* The fact that this doesn't directly align with anything in the Material Guidelines, whereas a checkbox list is present in the guidelines and can also be accomplished CSS-only if desired

I've entered #1792 to pursue checkbox list and decommission multiselect.
